### PR TITLE
Release/141t emr fixes

### PR DESCRIPTION
**Pull request by tomz**
PR for the three fixes from the EMR team.  the "Fix the varchar colums as parition column issue" is a simpler version of the PR made to the Facebook's master branch https://github.com/prestodb/presto/pull/5246.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 1]
```diff
@@ -479,9 +479,12 @@ else if (!partitionColumnNames.contains(name)) {
         table.setPartitionKeys(partitionColumns);
         table.setSd(sd);
 
-        PrivilegeGrantInfo allPrivileges = new PrivilegeGrantInfo("all", 0, tableOwner, PrincipalType.USER, true);
+        PrivilegeGrantInfo selectPrivilege = new PrivilegeGrantInfo("select", 0, tableOwner, PrincipalType.USER, true);
```
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:482]: @amrutagokhale can you confirm it looks good?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java [hunk 2]
```diff
@@ -439,7 +439,7 @@ public static NullableValue parsePartitionValue(String partitionName, String val
                 return NullableValue.of(DOUBLE, parseDouble(value));
             }
 
-            if (HIVE_STRING.equals(hiveType)) {
+            if (HIVE_STRING.equals(hiveType) || hiveType.toString().toLowerCase().startsWith("varchar")) {
```
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: Use `StandardTypes.VARCHAR` and `toLowerCase(Locale.ENGLISH)` for consistency with rest of codebase.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 1 above)
Comment (amrutagokhale) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:482]: @tomz can you please explain the reason behind replacing "all" privileges with individual ones? The owner of the table gets all the privileges (with grant option) under SQL standard, so why is this needed?

Comment (tomz) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:482]: @amrutagokhale under Hive SQL standard auth, ALL no longer exists, it's broken up into four separate priviledges(SELECT/INSERT/UPDATE/DELETE).

Comment (amrutagokhale) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:482]: I see. Was this a recent change? I couldn't find any mention of this here: https://cwiki.apache.org/confluence/display/Hive/SQL+Standard+Based+Hive+Authorization

Comment (tomz) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:482]: It could be after the Presto auth is implemented, the docs says "ALL PRIVILEGES – gives all privileges (gets translated into all the above privileges)" - it means that literally, when creating a table in Hive, it gets the four separate priviledges(S/I/U/D). Table created in Presto gets one 'ALL' privilege which does not exist anymore, and the table can not be read by Hive(throws exception about ALL not found or exist).

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java (hunk 2 above)
Comment (tomz) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: @maciejgrzybek does StandardTypes.VARCHAR translated to varchar?  And would hiveType.toString().toLowerCase(Locale.ENGLISH).startsWith(StandardTypes.VARCHAR) be equavalent to hiveType.toString().toLowerCase().startsWith("varchar"), we need to account for both varchar and varchar(n) cases.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 1 above)
Comment (amrutagokhale) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:482]: Perhaps you misunderstood me. I wanted to know when was this change of _removing_ ALL privileges introduced in Hive. Can you please point me to a Hive document or release notes that mention about _not_ supporting ALL privileges? Or has this not yet been documented in Hive, but has been implemented?

**Comment by amrutagokhale** [comment]
Changes related to privileges look good.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java (hunk 2 above)
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: If you want to use `varchar` and `varchar(x)` what not check `HiveType.getTypeSignature().getBase()`?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 1 above)
Comment (tomz) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java:482]: ah I see, I didn't look for that detail. Our customers experienced the issue and asked for a fix. Will let you know if I find out more about it.

**Comment by tomz** [comment]
@maciejgrzybek can you check the last commit? - addressed your comment about using StandardType.VARCHAR

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java [hunk 3]
```diff
@@ -439,7 +439,7 @@ public static NullableValue parsePartitionValue(String partitionName, String val
                 return NullableValue.of(DOUBLE, parseDouble(value));
             }
 
-            if (HIVE_STRING.equals(hiveType) || hiveType.toString().toLowerCase().startsWith("varchar")) {
+            if (HIVE_STRING.equals(hiveType) || hiveType.getTypeSignature().getBase().equals(VARCHAR.toString())) {
```
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: This won't work with parametrized VARCHAR. Use `hiveType.getTypeSignature().getBase().equals(VARCHAR.getTypeSignature().getBase())` instead.

**Comment by maciejgrzybek** [comment]
Looks good % small change. @tomz back to you

**Comment by tomz** [comment]
@maciejgrzybek thanks, done

**Comment by maciejgrzybek** [comment]
@tomz I will merge it as soon as it passes build.

**Comment by maciejgrzybek** [comment]
Merged, thanks!
